### ISSUE TITLE
Will this sleep() before a retry block the broker application?

### ISSUE THREAD

**Issue by matheuscscp** [emqx/emqx-web-hook#140]
https://github.com/emqx/emqx-web-hook/blob/45b4a55de8cd240e7698b0d45cff1ebe811db769/src/emqx_web_hook.erl#L289

This sleep() runs before another request attempt is sent. Will this sleep() block the entire broker, or the webhook plugin runs in a separate thread?

**Comment by HJianBo** [comment]
Nope. It does not block the entire broker. These lines running in each connection process.

Don't worry about that :-)

**Comment by matheuscscp** [comment]
Each connection has a whole process for it?

**Comment by HJianBo** [comment]
Yes. it is
